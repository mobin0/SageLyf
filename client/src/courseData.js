var courseData = [{"title":"Team-oriented discrete challenge","sectionNumber":1,"entries":[{"name":"Talk by Dr. Kolby Welch","duration":29,"entryNumber":1},{"name":"Talk by Liana Mayer","duration":55,"entryNumber":2},{"name":"Talk by Ulices Braun","duration":16,"entryNumber":3},{"name":"Talk by Raul Klein","duration":66,"entryNumber":4},{"name":"Talk by Ana Hudson","duration":33,"entryNumber":5},{"name":"Talk by Reva Treutel","duration":6,"entryNumber":6},{"name":"Talk by Rigoberto Dietrich II","duration":23,"entryNumber":7},{"name":"Talk by Vicky Daugherty","duration":74,"entryNumber":8}]},{"title":"Reverse-engineered attitude-oriented service-desk","sectionNumber":2,"entries":[{"name":"Talk by Katlynn Davis","duration":52,"entryNumber":1},{"name":"Talk by Mr. Lavern Weissnat","duration":84,"entryNumber":2},{"name":"Talk by Miss Ila Ondricka","duration":4,"entryNumber":3}]}]

export default courseData;