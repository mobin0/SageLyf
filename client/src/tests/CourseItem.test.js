import React from 'react';
import Item from "./../components/CourseItem.jsx";
import renderer from "react-test-renderer"
import { cleanup, fireEvent, waitForElement } from 'react-testing-library';
import {shallow, mount, render, configure} from 'enzyme';
import 'jest-dom/extend-expect';
import Adapter from 'enzyme-adapter-react-16';
import courseData from './../../../db/test.js'


configure({ adapter: new Adapter() });

afterEach(cleanup);

const testCase = courseData[0]

describe('CourseItem', () => {
  let mockUpdateState = jest.fn()
  const CourseItem = shallow(<Item courseItem = {testCase} updateCourseItemStates = {mockUpdateState}/>)

  it ('should have same snapshot', () => {
  expect(CourseItem).toMatchSnapshot()
  })

  it ('should load', () => {
 	expect(CourseItem.exists()).toBeTruthy()
  })

  it ('should have called update function', () => {
 	expect(mockUpdateState).toHaveBeenCalled()
  })

  it ('should have a + sign', () => {
 	expect(CourseItem.find('.plus-sign').text()).toBe("+")
  })

  it ('should have correct course title', () => {
  	expect(CourseItem.find('.course-title').text()).toBe(testCase.title)
  })

  it ('should have the correct hours', () => {
  	expect(CourseItem.state().hours).toBe("04")
  })

  it ('should have the correct minutes', () => {
  	expect(CourseItem.state().minutes).toBe("09")
  })

  it ('should have the course-lecture-length visible', () => {
 	expect(CourseItem.find('.course-lecture-length').length).toBe(1)
  })

  it ('should have the course entries hidden', () => {
 	expect(CourseItem.find('.course-entries').length).toBe(1)
  })

})