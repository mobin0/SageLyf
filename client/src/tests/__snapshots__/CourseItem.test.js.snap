// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseItem should have same snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseItem
    courseItem={
      Object {
        "entries": Array [
          Object {
            "duration": 249,
            "entryNumber": 1,
            "name": "Talk by Agustin Murray",
          },
        ],
        "sectionNumber": 1,
        "title": "Progressive 24/7 Graphical User Interface",
      }
    }
    updateCourseItemStates={
      [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="course-item"
          onClick={[Function]}
        >
          <span
            className="plus-sign"
          >
            +
          </span>
          <span
            className="course-title"
          >
            Progressive 24/7 Graphical User Interface
          </span>
          <span
            className="course-lecture-length"
          >
            1 lectures
          </span>
          <span
            className="course-lecture-duration"
          >
            04:09
          </span>
        </div>,
        <div
          className="course-entries"
          id="section1"
        >
          <Entries
            entry={
              Object {
                "duration": 249,
                "entryNumber": 1,
                "name": "Talk by Agustin Murray",
              }
            }
            sectionNumber={1}
          />
        </div>,
      ],
      "className": "course-item-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="plus-sign"
            >
              +
            </span>,
            <span
              className="course-title"
            >
              Progressive 24/7 Graphical User Interface
            </span>,
            <span
              className="course-lecture-length"
            >
              1 lectures
            </span>,
            <span
              className="course-lecture-duration"
            >
              04:09
            </span>,
          ],
          "className": "course-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "+",
              "className": "plus-sign",
            },
            "ref": null,
            "rendered": "+",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Progressive 24/7 Graphical User Interface",
              "className": "course-title",
            },
            "ref": null,
            "rendered": "Progressive 24/7 Graphical User Interface",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1 lectures",
              "className": "course-lecture-length",
            },
            "ref": null,
            "rendered": "1 lectures",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "04:09",
              "className": "course-lecture-duration",
            },
            "ref": null,
            "rendered": "04:09",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Entries
              entry={
                Object {
                  "duration": 249,
                  "entryNumber": 1,
                  "name": "Talk by Agustin Murray",
                }
              }
              sectionNumber={1}
            />,
          ],
          "className": "course-entries",
          "id": "section1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "{\\"name\\":\\"Talk by Agustin Murray\\",\\"duration\\":249,\\"entryNumber\\":1}",
            "nodeType": "class",
            "props": Object {
              "entry": Object {
                "duration": 249,
                "entryNumber": 1,
                "name": "Talk by Agustin Murray",
              },
              "sectionNumber": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="course-item"
            onClick={[Function]}
          >
            <span
              className="plus-sign"
            >
              +
            </span>
            <span
              className="course-title"
            >
              Progressive 24/7 Graphical User Interface
            </span>
            <span
              className="course-lecture-length"
            >
              1 lectures
            </span>
            <span
              className="course-lecture-duration"
            >
              04:09
            </span>
          </div>,
          <div
            className="course-entries"
            id="section1"
          >
            <Entries
              entry={
                Object {
                  "duration": 249,
                  "entryNumber": 1,
                  "name": "Talk by Agustin Murray",
                }
              }
              sectionNumber={1}
            />
          </div>,
        ],
        "className": "course-item-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="plus-sign"
              >
                +
              </span>,
              <span
                className="course-title"
              >
                Progressive 24/7 Graphical User Interface
              </span>,
              <span
                className="course-lecture-length"
              >
                1 lectures
              </span>,
              <span
                className="course-lecture-duration"
              >
                04:09
              </span>,
            ],
            "className": "course-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+",
                "className": "plus-sign",
              },
              "ref": null,
              "rendered": "+",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Progressive 24/7 Graphical User Interface",
                "className": "course-title",
              },
              "ref": null,
              "rendered": "Progressive 24/7 Graphical User Interface",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1 lectures",
                "className": "course-lecture-length",
              },
              "ref": null,
              "rendered": "1 lectures",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "04:09",
                "className": "course-lecture-duration",
              },
              "ref": null,
              "rendered": "04:09",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Entries
                entry={
                  Object {
                    "duration": 249,
                    "entryNumber": 1,
                    "name": "Talk by Agustin Murray",
                  }
                }
                sectionNumber={1}
              />,
            ],
            "className": "course-entries",
            "id": "section1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "{\\"name\\":\\"Talk by Agustin Murray\\",\\"duration\\":249,\\"entryNumber\\":1}",
              "nodeType": "class",
              "props": Object {
                "entry": Object {
                  "duration": 249,
                  "entryNumber": 1,
                  "name": "Talk by Agustin Murray",
                },
                "sectionNumber": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
